syntax = "proto3";

package Block;

message Block {
    BlockHeader BlockHeader = 1;
    Transactions Transactions = 2;
}   

message BlockHeader {
    uint32 block_size = 1;
    bytes previous_block_hash = 2;
    bytes merkle_root_hash = 3;
    uint32 timestamp = 4;
    uint32 nonce = 5;
}

message Transactions {
    repeated bytes user_transaction = 1;
}
